nexus/tests/integration_tests/unauthorized.rs

```diff
@@ -54,6 +55,24 @@ async fn test_unauthorized(cptestctx: &ControlPlaneTestContext) {
     let client = &cptestctx.external_client;
     let log = &cptestctx.logctx.log;
 
+    // Run a httptest server
```

Comment:
It seems like this is functioning as a mock...something?  Some public-facing image service that we'd be running?

Would it make sense to up this mocok as part of setting up the ControlPlaneTestContext, similar to the way we set up a simulated Sled Agent, CockroachDB, Clickhouse, Oximeter, and Crucible?

Reply:
Right now it's only for something to fire HEAD requests against. This reasoning makes sense to me though because basically every image integration test built one, but when I did this I saw a lot of failures because tests were all trying to bind to the same address and port (which was statically set because DEMO_IMAGE_CREATE is static)

Reply:
Yeah, I think for all the other things that we start, we let it bind to any available port and then report that port to whatever uses it (rather than fixing it).

This is a good catch -- it seems like this could break spuriously if this test were run concurrently.

Reply:
To make this is more explicit, we do this for:

Clickhouse (bind to port zero): https://github.com/oxidecomputer/omicron/blob/3817f7edeadb9620086d695bff90e88c617a0b28/nexus/test-utils/src/lib.rs#L99

Nexus (use `ConfigDropshot`'s default, which is `127.0.0.1:0`):
https://github.com/oxidecomputer/omicron/blob/3817f7edeadb9620086d695bff90e88c617a0b28/nexus/test-utils/src/lib.rs#L105-L107
https://github.com/oxidecomputer/omicron/blob/3817f7edeadb9620086d695bff90e88c617a0b28/nexus/src/config.rs#L57-L60
https://docs.rs/dropshot/0.6.0/src/dropshot/config.rs.html#54-61

Etc.

Dropshot's `local_addr()` method makes it easy to access the "real" port that was allocated after the fact:
https://github.com/oxidecomputer/omicron/blob/3817f7edeadb9620086d695bff90e88c617a0b28/nexus/test-utils/src/lib.rs#L131